### ISSUE TITLE
About your paper.

### ISSUE THREAD

**Issue by guyuchao** [AttendAndExcite/Attend-and-Excite#9]
You write in Section 4, "The pre-trained CLIP text encoder prepends a specialized token 〈sot〉 to P indicating the start of the text. During the text encoding process, this token receives global information about the prompt. This leads to 〈sot〉 obtaining a high probability in the token distribution defined in At." 

However, due to the casual mask in CLIP training, the 〈sot〉 will not receive any information. Why this token receives global information about the prompt?

**Comment by AttendAndExcite** [comment]
Hi @guyuchao, thanks for your interest!
the causal mask by CLIP will only mask the padding tokens. I used the text encoder with stable diffusion to produce an example of the mask for you:
prompt: "a photograph of a dog"
tokens: <sot>, a, photograph, of, a, dog, <eos>, <pad>, ..., <pad>
the attention mask: [1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 0, 0]
therefore, <sot> is not masked in the text decoding :)

**Comment by guyuchao** [comment]
But as in the code, it uses the upper triangle mask. How do you get that mask?

https://github.com/huggingface/transformers/blob/92ce53aab859012f7714dae6d6fce7a7d701e75f/src/transformers/models/clip/modeling_clip.py#L715

**Comment by AttendAndExcite** [comment]
Hi, @guyuchao, thanks for your interest! Just saw what you meant in the code, this will be fixed in the next revision (however, this still does not contradict the fact that the \<sot\> token obtains a high attention value for SD.

**Comment by guyuchao** [comment]
Great. Actually it is true that <sot> token obtains a high attention value for SD.
